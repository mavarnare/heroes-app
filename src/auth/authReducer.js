import { types } from "../types/type";

// const state = {
//     name: 'Fernando',
//     logged: true,
// }

export const authReducer = (state = {}, action) => {
    console.log('reducer - state:', state);
    console.log('reducer - action:', action);
    switch ( action.type ) {
        case types.login:
            return {
                ...action.payload,
                logged: true,
            };

        case types.logout:
            return {
                logged: false,
            };
    
        default:
            return state;;
    }
}