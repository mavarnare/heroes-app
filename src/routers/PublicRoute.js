import React, { useContext } from 'react'
import { Redirect, Route } from 'react-router';

export const PublicRoute = ({ isAuthenticated, ...rest }) => {

    return (
        isAuthenticated ? <Redirect to="/search" /> : <Route { ...rest } />
    )
}
