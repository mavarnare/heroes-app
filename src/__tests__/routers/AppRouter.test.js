import { mount } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import { AppRouter } from "../../routers/AppRouter";

describe('tests for <AppRouter />', () => {

    it('should show <LoginScreen /> if not authenticated', () => {
        
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false,
        }
    };
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('LoginScreen').exists() ).toBe( true );
        expect( wrapper.find('DashboardRoutes').exists() ).toBe( false );
    });

    it('should show <DashboardRoutes /> if authenticated', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Mavarnare',
            }
        };
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('LoginScreen').exists() ).toBe( false );
        expect( wrapper.find('DashboardRoutes').exists() ).toBe( true );
    });
    
});
