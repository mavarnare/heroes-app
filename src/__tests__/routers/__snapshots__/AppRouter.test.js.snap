// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <AppRouter /> should show <DashboardRoutes /> if authenticated 1`] = `
<div>
  Array [
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <div
        className="navbar-collapse"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/marvel"
            onClick={[Function]}
          >
            Marvel
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            aria-current="page"
            className="nav-item nav-link active"
            href="/search"
            onClick={[Function]}
            style={Object {}}
          >
            Search
          </a>
        </div>
      </div>
      <div
        className="navbar-collapse collapse w-100 order-3 dual-collapse2"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <span
            className="nav-item nav-link text-info"
          >
            Mavarnare
          </span>
          <button
            className="nav-item nav-link btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </ul>
      </div>
    </nav>,
    <div
      className="container mt-2"
    >
      <div>
        <h1>
          Search Screen
        </h1>
        <hr />
        <div
          className="row"
        >
          <div
            className="col-5"
          >
            <h4>
              Search Form
            </h4>
            <hr />
            <form
              onSubmit={[Function]}
            >
              <input
                autoComplete="off"
                className="form-control"
                name="searchText"
                onChange={[Function]}
                placeholder="Find your hero"
                type="text"
                value=""
              />
              <button
                className="btn m-1 btn-block btn-outline-primary"
                type="submit"
              >
                Search...
              </button>
            </form>
          </div>
          <div
            className="col-7"
          >
            <h4>
              Results
            </h4>
            <hr />
            <div
              className="alert alert-info"
            >
              Search a hero
            </div>
          </div>
        </div>
      </div>
    </div>,
  ]
</div>
`;

exports[`tests for <AppRouter /> should show <LoginScreen /> if not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
